pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -135,10 +136,13 @@ func (d *namespacedResourcesDeleter) Delete(nsName string) error {
 	// there may still be content for us to remove
 	estimate, err := d.deleteAllContent(namespace.Name, *namespace.DeletionTimestamp)
 	if err != nil {
+		d.updateStatusCondition(namespace.Name, err)
```

Comment:
how often does this happen? Seconds? Milliseconds? With exponential backoff?

Reply:
You need a code path that will remove the error condition if it eventually successfully deletes the thing, I think?

Reply:
at the moment, if the status condition no longer applies then the namespace is deleted. But I would like to re-think my proposed implementation based on your comments and after that a delete path will be probably needed